src/Accelerators/NNPA/Transform/ZLow/ZLowRewrite.cpp

```diff
@@ -100,6 +112,330 @@ class StickViewUnstickRemovalPattern : public OpRewritePattern<ZLowStickOp> {
+    for (AffineStoreOp storeOp : storeOps) {
+      Value storeMemref = storeOp.getMemref();
+      Value storeValue = storeOp.getValue();
+      ZLowStickOp myStickOp = StoreOpStickOpMap[storeOp];
+      Value stickMemref = myStickOp.getOut();
+
+      // Move stickMemref's AllocOp up before affine.for so that it
+      // dominates its uses. A good place is just after storeMemref's AllocOp.
+      //
+      // Get AllocOps that allocated storeMemref and stickMemref.
+      Operation *storeAllocOp = storeMemref.getDefiningOp();
+      Operation *stickAllocOp = stickMemref.getDefiningOp();
+      // stickAllocOp should be after storeAllocOp, since dimensions come from
+      // storeAllocOp according to the definition of zlow.stick.
+      Operation *justMovedOp = nullptr;
+      // Move AllocOp's operands first.
+      for (unsigned i = 0; i < stickAllocOp->getNumOperands(); ++i) {
+        Value oprd = stickAllocOp->getOperand(i);
+        if (isa<BlockArgument>(oprd))
+          continue;
+        Operation *opToMove = oprd.getDefiningOp();
+        // Do not move, it is potentially used by storeAllocOp and it is a good
+        // place already.
+        if (opToMove->isBeforeInBlock(storeAllocOp))
+          continue;
+        if (justMovedOp)
+          opToMove->moveAfter(justMovedOp);
+        else
+          opToMove->moveAfter(storeAllocOp);
+        justMovedOp = opToMove;
+      }
+      // Move AllocOp.
+      if (justMovedOp)
+        stickAllocOp->moveAfter(justMovedOp);
+      else
+        stickAllocOp->moveAfter(storeAllocOp);
+
+      // Replace storeOp.
+      OpBuilder::InsertionGuard guard(rewriter);
+      rewriter.setInsertionPointAfter(storeOp);
+      // This DummyOp is used to make the intermediate generated code valid. It
+      // will be removed automatically via canonicalization.
+      Value dummyConverter =
+          rewriter.create<ZLowDummyOp>(loc, stickifiedElementType, storeValue);
+      // Clone storeOp with new Memref, Value, and Indices.
+      IRMapping operandMap;
+      operandMap.map(storeOp.getMemref(), stickMemref);
+      operandMap.map(storeOp.getValue(), dummyConverter);
+      // Permute indices in case of NCHW layout.
+      if (isNCHWLayout) {
+        // for zlow.stick: input is NCHW, output is NHWC.
+        ValueRange NCHWIndices = storeOp.getIndices();
+        SmallVector<Value, 4> NHWCIndices;
+        NHWCIndices.emplace_back(NCHWIndices[0]); // N
+        NHWCIndices.emplace_back(NCHWIndices[2]); // H
+        NHWCIndices.emplace_back(NCHWIndices[3]); // W
+        NHWCIndices.emplace_back(NCHWIndices[1]); // C
+        operandMap.map(NCHWIndices, NHWCIndices);
+      }
+      rewriter.clone(*storeOp.getOperation(), operandMap);
+      rewriter.eraseOp(storeOp);
+    }
+
+    // Remove ZLowUnstickOp.
+    rewriter.eraseOp(unstickOp);
+    // Copy ZLowStickOp to the removableStickOps. We cannot remove it now
+    // because there are potentially other AffineStoreOps connecting to it.
```

Comment:
connecting -> using? Clearer on use/def relation.

Reply:
Fixed this. Thanks!